applications/FluidDynamicsApplication/custom_elements/stokes_3D_twofluid.h

```diff
@@ -887,6 +872,41 @@ class Stokes3DTwoFluid
         data.stress[5] =  c2*strain[5];
     }
 
+    //compute strain
+
+    Vector CalculateStrain()
+    {
+        const unsigned int NumNodes = 4;
+        const unsigned int Dim = 3;
+        const unsigned int strain_size = 6;
+
+        //struct to pass around the data
+        element_data<NumNodes,Dim> data;
+
+        //getting data for the given geometry
+        double Volume;
+        GeometryUtils::CalculateGeometryData(GetGeometry(), data.DN_DX, data.N, Volume);
+
+        for (unsigned int i = 0; i < NumNodes; i++)
+        {
+            const array_1d<double,3>& vel = GetGeometry()[i].FastGetSolutionStepValue(VELOCITY);
+            for(unsigned int k=0; k<Dim; k++)
+                data.v(i,k)   = vel[k];
+        }
+
+        const BoundedMatrix<double,NumNodes,Dim>& v = data.v;
+        const BoundedMatrix<double,NumNodes,Dim>& DN = data.DN_DX;
```

Comment:
Why do not use these directly instead of creating the element_data?

Reply:
I was following the original design , so just copied the procedure.
you're right, I'll change this